src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -24,8 +24,8 @@ public class Queue<T> : IEnumerable<T>,
         IReadOnlyCollection<T>
     {
         private T[] _array;
-        private int _head;       // First valid element in the queue
-        private int _tail;       // Last valid element in the queue
+        private int _head;       // Index of the first valid element in the queue.
+        private int _tail;       // Index of the first invalid element (or index of the last valid + 1) in the queue.
```

Comment:
Thanks for the tweak.  Technically there are times when both of these comments are still wrong.  For example, if the queue is empty, _head does not point to the index of the first valid element, as there is no valid element.  And if the queue's count equals the array's size, then _tail is also going to have the same value as _head, pointing to a valid element.  But maybe that's nit-picking?  Regardless, your change is an improvement over what's there, though if I were writing these from scratch I'd probably have written for _head "The index from which to dequeue if the queue isn't empty" and for _tail "The index at which to enqueuer if the queue isn't full", or something like that.  If you stick with what you changed this to, I'd suggest changing "valid + 1" to be "(valid + 1) % _array.Length", since it wraps around.